src/php/iterators/php_htable_iterator.h

```diff
@@ -10,6 +10,7 @@ typedef struct ds_htable_iterator {
     uint32_t                 position;
     ds_htable_bucket_t      *bucket;
     ds_htable_t             *table;
+    zval                    *obj;
```

Comment:
Can do either easily enough. Any preference for zval vs zend_object* here or is the result the same? The current patch passed the tests without leaks but I see what you're saying here. Will need to do the same with the sequence iterators too.